### PR TITLE
feat: OZ support

### PR DESCRIPTION
**Pull request by mds1**
Adds OZ support to Seatbelt. Kicked off CI with ENS and Unlock as the two test OZ Governors: https://github.com/Uniswap/governance-seatbelt/actions/runs/2952226908

Remaining tasks:
- [x] Fix any failures in that CI run (seems like lots of 404s, unsure yet why, things generally worked when testing locally)
- [x] Test a few local sims for ENS and Unlock
- [x] Be consistent with BigNumber vs. bigint usage
- [x] Remove unused imports
- [x] Renamed `compound` to `bravo`
- [x] Make sure we never access governor specific logic directly (e.g. we currently use `getBravoSlots` in `check-state-changes.ts`)

### CODE REVIEW

File: utils/contracts/governor.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+import { GovernorType, ProposalEvent, ProposalStruct } from '../../types'
+
+// --- Exported methods ---
+export async function inferGovernorType(address: string): Promise<'oz' | 'bravo'> {
+  const abi = ['function initialProposalId() external view returns (uint256)']
+  const governor = new Contract(address, abi, provider)
+
+  try {
+    // If the `initialProposalId` function exists, and the initial proposal was a "small" value,
+    // it's overwhelmingly likely this is. OZ style governors use a hash as the proposal IDs so IDs
```
Comment (wildmolasses) [inline utils/contracts/governor.ts:19]:
```suggestion
    // it's overwhelmingly likely this is GovernorBravo. OZ style governors use a hash as the proposal IDs so IDs
```

File: utils/contracts/governor.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        [targets, values, calldatas, keccak256(toUtf8Bytes(description))]
+      )
+    )
+  )
+}
+
+// Returns the identifier of an operation containing a batch of transactions.
+// For OZ governors, predecessor is often zero and salt is often description hash.
+// This is only intended to be used with OZ governors.
+export function hashOperationBatch(
```
Comment (wildmolasses) [inline utils/contracts/governor.ts:152]: nit: consider, feel free to ignore!

```suggestion
export function hashOperationBatchOZ(
```

File: types.d.ts [hunk 3]
```diff
@@ -3,10 +3,13 @@ import { ContractTransaction } from '@ethersproject/contracts'
 import { JsonRpcProvider } from '@ethersproject/providers'
 
 // --- Simulation configurations ---
+export type GovernorType = 'oz' | 'bravo'
```
Comment (wildmolasses) [inline types.d.ts:6]: generally we might prefer enum to string here, but given the small size of the codebase i dont suggest a refactor

**Review by wildmolasses — APPROVED** [review]
ran reports on my machine, and they look great with no apparent errors. here are a few small comments, otherwise lgtm

File: types.d.ts [hunk 4]
```diff
@@ -52,10 +55,12 @@ export type ProposalActions = [
 
 export interface ProposalStruct {
   id: BigNumber
-  proposer: string
+  proposer?: string
   eta: BigNumber
-  startBlock: BigNumber
-  endBlock: BigNumber
+  startBlock?: BigNumber // Compound governor
```
[the remark this answers is not in the document]
Comment (mds1) [reply types.d.ts:60]: Ah nice that's definitely a cleaner approach. I'll leave this for now and make that change if we ever add a third governor

File: .env.example [hunk 5]
```diff
@@ -2,6 +2,7 @@
 ETHERSCAN_API_KEY=yourEtherscanApiKey
 RPC_URL=yourMainnetRpcUrl
 TENDERLY_ACCESS_TOKEN=yourTenderlyAccessToken
+TENDERLY_USER=yourTenderlyOrgOrMeIfPersonal
 TENDERLY_PROJECT_SLUG=yourTenderlyProjectSlug
 
 # Only needed for GitHub Actions
```
[the remark this answers is not in the document]
Comment (mds1) [reply .env.example:8]: Yea I think the main use cases to consider are:
- Adding your DAO to the CI flow
- Running one or more sims locally for a single DAO
- Simulating new proposals locally before executing them on chain (OZ support for that will be in the next PR)

Given that, lmk if you think we should clarify anything

File: .env.example [hunk 6]
```diff
@@ -2,6 +2,7 @@
 ETHERSCAN_API_KEY=yourEtherscanApiKey
 RPC_URL=yourMainnetRpcUrl
 TENDERLY_ACCESS_TOKEN=yourTenderlyAccessToken
+TENDERLY_USER=yourTenderlyOrgOrMeIfPersonal
```
[the remark this answers is not in the document]
Comment (wildmolasses) [reply .env.example:7]: This has been resolved ✅

File: checks/check-slither.ts [hunk 7]
```diff
@@ -31,7 +32,8 @@ export const checkSlither: ProposalCheck = {
     // at the simulation block number, since the implementation may have changed since.
```
[the remark this answers is not in the document]
Comment (wildmolasses) [reply checks/check-slither.ts:32]: This has been resolved ✅

File: checks/check-solc.ts [hunk 8]
```diff
@@ -32,7 +33,8 @@ export const checkSolc: ProposalCheck = {
     // at the simulation block number, since the implementation may have changed since.
```
[the remark this answers is not in the document]
Comment (wildmolasses) [reply checks/check-solc.ts:33]: Resolved ✅

File: types.d.ts (hunk 3 above)
Comment (wildmolasses) [reply types.d.ts:7]: (implicitly resolved ✅ )

File: README.md [hunk 9]
```diff
@@ -29,31 +29,38 @@ to the `matrix` section of `.github/workflows/governance-checks.yaml`.
...
 GOVERNOR_ADDRESS=0x408ED6354d4973f66138C91495F2f2FCbd8724C3
 ```
 
-There are now two modes of operation:
+### Running Simulations
+
+There are two modes of operation:
 
 1. Run `yarn start` to simulate and run checks on all GovernorBravo proposals
 2. Create a file called `<analysisName>.sim.ts` and run a specific simulation with `SIM_NAME=analysisName yarn start`. See the `*.sim.ts` files in the `sims` folder for examples.
```
[the remark this answers is not in the document]
Comment (wildmolasses) [reply README.md]: Resolved ✅

File: README.md [hunk 10]
```diff
@@ -29,31 +29,38 @@ to the `matrix` section of `.github/workflows/governance-checks.yaml`.
 
 Note that currently only `GovernorBravo` style governors are supported.
```
[the remark this answers is not in the document]
Comment (wildmolasses) [reply README.md]: Resolved ✅

File: types.d.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (wildmolasses) [reply types.d.ts:63]: (implicitly resolved ✅ )

File: .env.example (hunk 5 above)
[the remark this answers is not in the document]
Comment (wildmolasses) [reply .env.example]: Resolved ✅

File: README.md [hunk 11]
```diff
@@ -29,31 +29,38 @@ to the `matrix` section of `.github/workflows/governance-checks.yaml`.
...
 DAO_NAME=Uniswap
 GOVERNOR_ADDRESS=0x408ED6354d4973f66138C91495F2f2FCbd8724C3
 ```
 
-There are now two modes of operation:
+### Running Simulations
+
+There are two modes of operation:
 
 1. Run `yarn start` to simulate and run checks on all GovernorBravo proposals
```
[the remark this answers is not in the document]
Comment (wildmolasses) [reply README.md]: Resolved ✅

File: utils/contracts/governor.ts (hunk 1 above)
Comment (wildmolasses) [reply utils/contracts/governor.ts]: Resolved ✅

File: utils/contracts/governor.ts (hunk 2 above)
Comment (wildmolasses) [reply utils/contracts/governor.ts]: Resolved ✅

File: utils/contracts/governor.ts [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+import { keccak256 } from '@ethersproject/keccak256'
+import { defaultAbiCoder } from '@ethersproject/abi'
+import { provider } from '../clients/ethers'
+import { governorBravo, getBravoSlots } from './governor-bravo'
+import { governorOz, getOzSlots } from './governor-oz'
+import { timelock } from './timelock'
+import { GovernorType, ProposalEvent, ProposalStruct } from '../../types'
+
+// --- Exported methods ---
+export async function inferGovernorType(address: string): Promise<'oz' | 'bravo'> {
```
[the remark this answers is not in the document]
Comment (wildmolasses) [reply utils/contracts/governor.ts]: Resolved ✅

**Pull request merged**